wcs-functions.php

```diff
@@ -30,16 +30,18 @@
 
 /**
  * Check if a given object is a WC_Subscription (or child class of WC_Subscription), or if a given ID
- * belongs to a post with the subscription post type ('shop_subscription')
+ * belongs to a post with the subscription post type ('shop_subscription').
  *
  * @since  1.0.0 - Migrated from WooCommerce Subscriptions v2.0
+ *
+ * @param mixed $subscription A WC_Subscription object or post ID
  * @return boolean true if anything is found
  */
 function wcs_is_subscription( $subscription ) {
 
 	if ( is_object( $subscription ) && is_a( $subscription, 'WC_Subscription' ) ) {
 		$is_subscription = true;
-	} elseif ( is_numeric( $subscription ) && 'shop_subscription' == get_post_type( $subscription ) ) {
+	} elseif ( is_numeric( $subscription ) && 'shop_subscription' === WC_Data_Store::load( 'subscription' )->get_order_type( $subscription ) ) {
```

Comment:
Thinking out loud here. Is there any chance `WC_Data_Store` isn't defined or `WC_Data_Store::load( 'subscription' )` is not an object resulting a fatal error? I can't think of any.

Reply:
👍  I don't think so, as long as the `WCS_Orders_Table_Data_Store_Controller()` has been instantiated, which happens early in the `WC_Subscriptions_Core_Plugin` constructor.

https://github.com/Automattic/woocommerce-subscriptions-core/blob/9e0566089006283af18a70d6326252e5cc0b7219/includes/class-wc-subscriptions-core-plugin.php#L61